pkg/apis/migration/v1alpha1/migcluster_types.go

```diff
@@ -218,6 +220,23 @@ func (m *MigCluster) GetClusterSubdomain(c k8sclient.Client) (string, error) {
 	return clusterSubdomain, nil
 }
 
+// GetRegistryImage gets a MigCluster specific registry image from ConfigMap
+func (m *MigCluster) GetOperatorVersion(c k8sclient.Client) (string, error) {
+	client, err := m.GetClient(c)
+	if err != nil {
+		return "", err
+	}
+	clusterConfig, err := m.GetClusterConfigMap(client)
```

Comment:
In this situation, you need to grab the client for this MigCluster first, since the `ClusterConfigMap` is located in etcd on the MigCluster rather than on the controller cluster (these are equivalent when MigCluster == host).

See `MigCluster.GetClient()`

```
# Pass the reconciler client to MigCluster.GetClient
clusterClient, err := m.GetClient(c)
# Use the clusterClient to get the configmap from the (possibly remote) MigCluster
clusterConfig, err := m.GetClusterConfigMap(clusterClient)
```

Reply:
@djwhatle I thought I was grabbing the client on line 225 just above the error check?